HW3 - COMMAND LINE. DEBUGGING. ERROR HANDLING / FILE SYSTEM AND STREAMS

wenditer opened the pull request:
- add CLI for executing custom functions from streams.js

wenditer committed:
HW3 - COMMAND LINE. DEBUGGING. ERROR HANDLING / FILE SYSTEM AND STREAMS

- add CLI for executing custom functions from streams.js

abruzgin commented on src/utils/streams.js:
```diff
@@ -0,0 +1,158 @@
+import reverseString from "reverse-string"
+import fs from "fs"
+import csv from "csvtojson"
+import Deque from "collections/deque";
```
Try to avoid usage of `collection` library, use native `Array, Set, Map` instead. If you're not familiar with them already, please check this
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Set
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Map

abruzgin commented on src/utils/streams.js:
```diff
@@ -0,0 +1,158 @@
+
+function reverse() {
+  rl.question("Please provide a string for reversing: ", (answer) => {
+    console.log(`Reversed: ${reverseString(answer)}`);
```
Remove `reverseString` package. There's a simple solution in JS
```
answer.split('').reverse().join('');
```

wenditer replied:
it's not so straight-forward and clear actually, but yes, pretty simple =) will do.

abruzgin commented on src/utils/streams.js:
```diff
@@ -0,0 +1,158 @@
+function convertFromFile(filePath) {
+  validateFilePath(filePath, 'convertFromFile');
+  fs.createReadStream(filePath)
+  .on('error', handleFileReadError)
```
Just the indentation issue

abruzgin commented on src/utils/streams.js:
```diff
@@ -0,0 +1,158 @@
+    var cssFiles = new Deque(filesFromFS)
+      .filter(file => file.endsWith('.css'))
+      .filter(file => file !== 'bundle.css')
+      .map(fileName => {return dirPath + fileName;});
```
In arrow functions if you're handling only return value immediately, you can just put
`.map(fileName => dirPath + fileName);`

wenditer committed:
HW3 - COMMAND LINE. DEBUGGING. ERROR HANDLING / FILE SYSTEM AND STREAMS

- removed reverseString package dependency
- fixed indentation
- DeQueue pop() was replaced with shift() to make it work as queue
- simplified arrow function usage - removed curly braces

wenditer committed:
HW3 - COMMAND LINE. DEBUGGING. ERROR HANDLING / FILE SYSTEM AND STREAMS

- refactoring: move functionality to different modules to achieve single responsibility and high cohesion
